packages/workspace/src/browser/quick-open-workspace.ts

```diff
@@ -30,10 +30,13 @@ export class QuickOpenWorkspace implements QuickOpenModel {
     @inject(MessageService) protected readonly messageService: MessageService;
     protected readonly workspace: string | undefined;
 
-    open(workspaces: string[]): void {
+    async open(workspaces: string[]): Promise<void> {
         this.items = [];
+        const home = await this.workspaceService.home;
+        const platform = await this.workspaceService.platform;
```

Comment:
https://github.com/theia-ide/theia/blob/a821f30a4961913bd09fc5793bf2cf9114d525a6/packages/core/src/common/os.ts#L29

Reply:
It needs to get the backend's platform. But still should not be added to `workspaceService`. Can't we detect that from the URI somehow?

Reply:
@svenefftinge @akosyakov perhaps I can create a service in core named `HostService` which provides methods related to information from the host server, and can be further updated to include any necessary helper methods.

Example: `getPlatform(), isHostWindows(), isHostMac(), getHomedir(), ...`